applications/dashboard/controllers/api/UsersApiController.php

```diff
@@ -598,6 +598,35 @@ public function post_requestPassword(array $body) {
         $this->validateModel($this->userModel, true);
     }
 
+    /**
+     * @param $id
+     * @param array $body
+     * @return array $result The response body
+     * @throws Exception
+     * @throws NotFoundException if unable to find the user.
+     */
+    public function post_confirmEmail ($id, array $body) {
+        $this->permission('Garden.Users.Edit');
+
+        $in = $this->schema( ['emailKey:s' => 'key generated on registration'])->setDescription('Confirm new user email upon registration');
```

Comment:
The line is a little long, so the formatting could use some breathing room. Maybe something like...

```php
$in = $this->schema([
    'emailKey:s' => 'key generated on registration'
])->setDescription('Confirm new user email upon registration');
```

Reply:
I think these descriptions should be more generic. The vast majority of uses are going to be for new users, but that doesn't mean that's all it can be used for. Some suggestions:

1. emailKey - Email confirmation code
1. Description - Confirm a users current email address by using a confirmation code

Reply:
I'm thinking out-loud here, but maybe we can rename the "email key" parameter to "confirmation code", "code" or something similar. We use "email key" internally, but we have a chance to address renaming some things to a more user-friendly version as we expose them via the API. The word "key" strikes me as an index, which this isn't. To me, this is a randomly-generated "code".

Nothing that necessarily has to change, just opening it up for discussion.